Source/santametricservice/Writers/SNTMetricHTTPWriter.m

```diff
@@ -43,22 +43,21 @@ - (BOOL)write:(NSArray<NSData *> *)metrics toURL:(NSURL *)url error:(NSError **)
 
   _authSession.serverHostname = url.host;
   NSURLSession *_session = _authSession.session;
+  NSURLSessionDataTask *task;
 
   dispatch_group_t requests = dispatch_group_create();
-
-  [metrics enumerateObjectsUsingBlock:^(id value, NSUInteger index, BOOL *stop) {
+  for (NSData *metric in metrics) {
     dispatch_group_enter(requests);
 
-    request.HTTPBody = (NSData *)value;
-    NSURLSessionDataTask *task = [_session
+    request.HTTPBody = (NSData *)metric;
+    task = [_session
       dataTaskWithRequest:request
         completionHandler:^(NSData *_Nullable data, NSURLResponse *_Nullable response,
                             NSError *_Nullable err) {
           if (err != nil) {
             _blockError = err;
-            *stop = YES;
           } else if (response == nil) {
-            *stop = YES;
+            _blockError = nil;
```

Comment:
This will overwrite any previous error from other requests

Reply:
Also should ensure that the assignments are atomic (for both here and the assignment above when in the case `err != nil`).